src/plugins/workspace/server/routes/index.ts

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { schema } from '@osd/config-schema';
+
+import { CoreSetup, Logger } from '../../../../core/server';
+import { IWorkspaceDBImpl } from '../types';
+
+const WORKSPACES_API_BASE_URL = '/api/workspaces';
+
+const workspaceAttributesSchema = schema.object({
+  description: schema.maybe(schema.string()),
+  name: schema.string(),
+  features: schema.maybe(schema.arrayOf(schema.string())),
+  color: schema.maybe(schema.string()),
+  icon: schema.maybe(schema.string()),
+  defaultVISTheme: schema.maybe(schema.string()),
+});
+
+export function registerRoutes({
+  client,
+  logger,
+  http,
+}: {
+  client: IWorkspaceDBImpl;
+  logger: Logger;
+  http: CoreSetup['http'];
+}) {
+  const router = http.createRouter();
+  router.post(
+    {
+      path: `${WORKSPACES_API_BASE_URL}/_list`,
+      validate: {
+        body: schema.object({
+          search: schema.maybe(schema.string()),
+          sortOrder: schema.maybe(schema.string()),
+          perPage: schema.number({ min: 0, defaultValue: 20 }),
```

Comment:
Oh, didn't know that. Do you mean that we should make `perPage` as a uiSettings that user can customize in advanced setttings?

But I think it is good to use some hard-coded default value for now because

Reply:
Yeah! Which would be set within the public space of the code so yeah it's a good idea to keep this configure. It was more so a thing to add to your backlog.

Reply:
Sure, will do that later.